### PR TITLE
docs(core): Extended the signal docs to properly indicate how to use …

### PR DESCRIPTION
**Pull request by ngehlert**
…with the @if controlflow and for component inputs

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by wartab** [comment]
`@if(mySignal)` doesn't return anything. mySignal is just truthy and the condition becomes valid as a result.

**Comment by JeanMeche** [comment]
I'm wondering if this is something that should be caught and a warning/error raised by one of our extended diagnostic.

**Comment by wartab** [comment]
I don't think it should error. Even if it is dirty, I could see someone actually wanting to check the existence of a signal before accessing its value for example. 
fwiw [I requested at JetBrains](https://youtrack.jetbrains.com/issue/WEB-65471/Angular-Add-warning-when-signals-are-accidentally-attempted-to-be-read-without-the-function-call) that such usages of signals be considered warnings by default a while back and given it's been upvoted a few times, I think people tend to agree that it should raise some alarm.

**Comment by ngehlert** [comment]
yeah maybe I phrased it bad. but if the signal is defined `@if(mySignal)` will always result in the content being visible because the output of the expression is truthy even if the current value of the signal would be false.

**Comment by mauriziocescon** [comment]
This is actually valid code: 

```typescript
@Component({
  selector: 'app-root',
  standalone: true,
  template: `
   @if (example) {
     <div>Now defined to: {{ example() }}</div>
   } @else{
     <div>Undefined!</div>
   }
  `,
})
export class App implements OnInit {
  example: Signal<string> | undefined = undefined;

  ngOnInit(): void {
    setTimeout(() => (this.example = signal('10')), 5000);
  }
}
```

In general, assuming `example = signal(...)` I get the following in v18: 

[ERROR] NG4: This condition will always return true since this function is always defined. Did you mean to call it instead? [plugin angular-compiler]

**Comment by mauriziocescon** [comment]
It seems we're covered in all cases... 

https://stackblitz.com/edit/stackblitz-starters-fipz2g?file=src%2Fmain.ts

**Comment by ngehlert** [comment]
ok yeah. I was more interested in the case of `binding2 = input.required<string>();` but of course it's similar if the input even is a signal and you intentionally pass the signal itself forward instead of the menu.

Then I guess I need to update my Angular version to get those compiler errors 😅 

Do you want me to close the PR or do you think there could still be a use case having this information in the docs ready instead of understanding it implicitly if the error hits you?

**Comment by mauriziocescon** [comment]
Well, I'm just an ng-developer... I guess it's mostly a question for @JeanMeche! 

Strictly personal: I guess the average developer is expecting angular (via compiler or ts) to report as many errors as possible. 

I'd use doc-warnings mostly for common pitfalls (something like: it's designed in a specific way and we know you can get lost the first time you use it).

**Comment by ngehlert** [comment]
ok :D yeah that's one of the reason where this PR started of. We had now couple of times already that somebody was submitting a code change that had like an `@if (mySignal)` without the parentheses or used it in an input binding without it. And this caused bugs in the onChanges, because suddenly you getting a signal instead of a value 😅 

But if those issues are covered by the compiler in the next version that's fine for me :)

File: adev/src/content/guide/signals/inputs.md [hunk 1]
```diff
@@ -56,6 +56,15 @@ As with signals declared via `signal()`, you access the current value of the inp
 
 This access to the value is captured in reactive contexts and can notify active consumers, like Angular itself, whenever the input value changes.
 
+Using a signal to pass a value to your component you have to access the value in the same way.
+```angular-html
+<my-component [firstName]="firstName()"/>
```
Comment (JeanMeche) [inline adev/src/content/guide/signals/inputs.md:61]: I'm really wondering if this is necessary to add, as typechecking will throw an error if you don't invoke the signal.

**Comment by JeanMeche** [comment]
After a 2nd thought, I'd say this isn't worth adding to the docs. This more about template than signals. Most of it is covered by the compiler itself. 
Maybe we could introduce a extended diagnostic for conditions that are always truthy/null like in your case but that another topic.

**Pull request closed without merging**